napari/_app/_app.py

```diff
@@ -0,0 +1,59 @@
+from __future__ import annotations
+
+from functools import lru_cache
+from typing import Dict
+
+from app_model import Application
+
+from ._submenus import SUBMENUS
+from .actions._layer_actions import LAYER_ACTIONS
+from .injection._processors import PROCESSORS
+from .injection._providers import PROVIDERS
+
+
+class NapariApplication(Application):
+    def __init__(self) -> None:
+        # raise_synchronous_exceptions means that commands triggered via
+        # ``execute_command`` will immediately raise exceptions. Normally,
+        # `execute_command` returns a Future object (which by definition does not
+        # raise exceptions until requested).  While we could use that future to raise
+        # exceptions with `.result()`, for now, raising immediately should
+        # prevent any unexpected silent errors.  We can turn it off later if we
+        # adopt asynchronous command execution.
+        super().__init__('napari', raise_synchronous_exceptions=True)
+
+        self.injection_store.namespace = _napari_names  # type: ignore [assignment]
+        self.injection_store.register(
+            providers=PROVIDERS, processors=PROCESSORS
+        )
+
+        for action in LAYER_ACTIONS:
+            self.register_action(action)
+
+        self.menus.append_menu_items(SUBMENUS)
```

Comment:
I *love* how simple this is - is there a specific reason you prefer taking these names from the outer scope rather than passing them to the constructor?

Reply:
nothing all that consequential.  very minor thought is that we're only ever going to "construct" this thing once, so it doesn't really matter